### DISCUSSION TITLE
osu!lazer triangle effect feature

### DISCUSSION THREAD

**Discussion by Jacob1863** [ppy/osu#29162] [Q&A]
Hi! I am experiencing issues with performance in osu!lazer. I get like 20-30 FPS when triangles are in the Song Select.
I would really like a feature to add an option to enable or disable triangles effect! Here's a screenshot of osu!lazer lagging from the triangles effect. 

![performance-issue](https://github.com/user-attachments/assets/39ff033c-694d-4e90-9de2-1dc646ca6e8f)

**Comment by Joehuu** [comment]
The aim is to optimize the game as much as possible first before trying to remove effects. It might not even be the triangles effect that is lagging you.

I wonder what computer specs you have because the fps is pretty low.

**Reply by Jacob1863** [reply]
Integrated Graphics (Intel HD Graphics)
2GB DDR3 RAM
(4GB DDRL3 RAM max)
Intel Pentium N3710

Here you go.

**Reply by Jacob1863** [reply]
But my iPhone runs it just fine.
Except when it randomly overheats towards osu!lazer and others. Except for low end games

**Reply by Joehuu** [reply]
How much fps do you get on gameplay? I believe that is the most important part.

Otherwise, I believe using stable is the best bet for now if you want an overall smooth experience, considering the specs.

**Reply by Jacob1863** [reply]
Around 60-122 FPS

**Comment by peppy** [comment]
It's not going to be triangles. Probably textures if anything.

**Reply by Jacob1863** [reply]
Yeah. Probably the textures and so…
I did some optimizations though.
It did improve, but I still get some really big lag spikes.
